### PR TITLE
Verify + save received extended header

### PR DESCRIPTION
**Pull request by mariusmihaic**
## Reasoning behind the pull request
- Verify intercepted received extended header and save it
  
## Proposed changes
- In order to verify the received extended header, we use the incoming header processor. This will create incoming scrs + miniblocks from the received header + incoming events. If the hash of received data matches the computed one, then the proof is valid.
- Save will add incoming header once.

## Testing procedure
- 
- 
- 

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: cmd/sovereignnode/incomingHeader/incomingHeaderProcessor.go [hunk 1]
```diff
@@ -84,6 +85,15 @@ func (ihp *incomingHeaderProcessor) AddHeader(headerHash []byte, header sovereig
 	return nil
 }
 
+func (ihp *incomingHeaderProcessor) CreateExtendedHeader(header sovereign.IncomingHeaderHandler) (data.ShardHeaderExtendedHandler, error) {
```
Comment (sasurobert) [inline cmd/sovereignnode/incomingHeader/incomingHeaderProcessor.go:88]: need a comment

Comment (SebastianMarian) [reply cmd/sovereignnode/incomingHeader/incomingHeaderProcessor.go:88]: +1

File: process/interceptors/processor/sovereignHdrInterceptorProcessor.go [hunk 2]
```diff
@@ -16,50 +21,113 @@ type ArgsSovereignHeaderInterceptorProcessor struct {
...
+	}
+
+	computedHeaderHash, err := core.CalculateHash(hip.marshaller, hip.hasher, computedExtendedHeader)
+	if err != nil {
+		return err
+	}
+
+	if !bytes.Equal(computedHeaderHash, hash) {
+		return fmt.Errorf("%w, computed hash: %s, received hash: %s",
+			errors.ErrInvalidReceivedSovereignProof,
```
Comment (SebastianMarian) [inline process/interceptors/processor/sovereignHdrInterceptorProcessor.go:106]: It is proof or header?

File: cmd/sovereignnode/incomingHeader/incomingHeaderProcessor.go (hunk 1 above)
Comment (mariusmihaic) [reply cmd/sovereignnode/incomingHeader/incomingHeaderProcessor.go:88]: Right, missed it

File: process/interceptors/processor/sovereignHdrInterceptorProcessor.go (hunk 2 above)
Comment (mariusmihaic) [reply process/interceptors/processor/sovereignHdrInterceptorProcessor.go:106]: For now, it is a header. Will be a proof in the future. Changed the name, though

**Review by sasurobert — APPROVED** [review]

**Review by SebastianMarian — APPROVED** [review]

**Pull request merged**